src/hooks/useReportIDs.tsx

```diff
@@ -0,0 +1,135 @@
+    priorityMode: OnyxEntry<PriorityMode>;
+};
+
+type WithReportIDsContextProviderProps = OnyxProps & {
+    children: React.ReactNode;
+    currentReportIDForTests?: string;
+};
+
+type ReportIDsContextValue = {
+    orderedReportIDs: string[];
+};
+
+const ReportIDsContext = createContext<ReportIDsContextValue>({
+    orderedReportIDs: [],
+});
+
+function WithReportIDsContextProvider({
+    children,
+    chatReports,
+    betas,
+    policies,
+    allReportActions,
+    transactionViolations,
+    policyMembers,
+    priorityMode,
+    /**
+     * Only required to make unit tests work, since we
+     * explicitly pass the currentReportID in LHNTestUtils
+     * to SidebarLinksData, so this context doesn't have an
+     * access to currentReportID in that case.
+     *
+     * This is a workaround to have currentReportID available in testing environment.
+     */
+    currentReportIDForTests,
+}: WithReportIDsContextProviderProps) {
+    const currentReportIDValue = useCurrentReportID();
+    const derivedCurrentReportID = currentReportIDForTests ?? currentReportIDValue?.currentReportID;
+    const {activeWorkspaceID} = useActiveWorkspace();
+
+    const policyMemberAccountIDs = useMemo(() => getPolicyMembersByIdWithoutCurrentUser(policyMembers, activeWorkspaceID, getCurrentUserAccountID()), [activeWorkspaceID, policyMembers]);
+
+    const getOrderedReportIDs = useCallback(
+        (currentReportID?: string) =>
+            SidebarUtils.getOrderedReportIDs(
+                currentReportID ?? null,
+                chatReports,
+                betas ?? [],
+                policies,
+                priorityMode,
+                allReportActions,
+                transactionViolations,
+                activeWorkspaceID,
+                policyMemberAccountIDs,
+            ),
+        [chatReports, betas, policies, priorityMode, allReportActions, transactionViolations, activeWorkspaceID, policyMemberAccountIDs],
+    );
+
+    const orderedReportIDs = useMemo(() => getOrderedReportIDs(), [getOrderedReportIDs]);
+
+    // We need to make sure the current report is in the list of reports, but we do not want
+    // to have to re-generate the list every time the currentReportID changes. To do that
+    // we first generate the list as if there was no current report, then we check if
+    // the current report is missing from the list, which should very rarely happen. In this
+    // case we re-generate the list a 2nd time with the current report included.
+    const orderedReportIDsWithCurrentReport = useMemo(() => {
+        if (derivedCurrentReportID && !orderedReportIDs.includes(derivedCurrentReportID)) {
+            return getOrderedReportIDs(derivedCurrentReportID);
+        }
+        return orderedReportIDs;
+    }, [getOrderedReportIDs, derivedCurrentReportID, orderedReportIDs]);
+
+    const contextValue = useMemo(
+        () => ({
+            orderedReportIDs: orderedReportIDsWithCurrentReport,
```

Comment:
I understand that this is a context that holds the list of ordered report IDs.

Could you clarify the relationship between the current report ID and this list? Specifically, why isn't it simply orderedReportIDs? Is it because the ordered report ID list might not include the current report ID?